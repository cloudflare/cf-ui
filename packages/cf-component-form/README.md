# cf-component-form

> Cloudflare Form Component

## Installation

```sh
$ npm install cf-component-form
```

## Usage

```js
const React = require('react');
const {
  Form,
  FormHeader,
  FormFooter,
  FormFieldset,
  FormLabel,
  FormFieldError
} = require('cf-component-form');
const {Flex, FlexItem} = require('cf-component-flex');
const {Button} = require('cf-component-button');
const Input = require('cf-component-input');
const Select = require('cf-component-select');
const Textarea = require('cf-component-textarea');

class Component extends React.Component {
  // ...

  render() {
    return (
      <Form layout="horizontal" onSubmit={this.handleSubmit}>
        <FormHeader title="Feedback Form"/>

        <FormFieldset legend="Name">
          <Flex spacing="wide">
            <FlexItem>
              <FormLabel>First Name</FormLabel>
              <Input name="firstName" type="text"
                value={this.state.firstName}
                onChange={this.handleFirstNameChange}/>
            </FlexItem>
            <FlexItem>
              <FormLabel>Last Name</FormLabel>
              <Input name="lastName" type="text"
                value={this.state.lastName}
                onChange={this.handleLastNameChange}/>
            </FlexItem>
          </Flex>
        </FormFieldset>

        <FormFieldset legend="Type">
          <FormLabel hidden>Type</FormLabel>
          <Select
            options={[
              { value: 'suggestion', label: 'Suggestion' },
              { value: 'problem', label: 'Problem' }
            ]}
            value={this.state.type}
            onChange={this.handleTypeChange}/>
        </FormFieldset>

        <FormFieldset legend="Message">
          <FormLabel hidden>Message</FormLabel>
          <Textarea name="message"
            value={this.state.message}
            onChange={this.handleMessageChange}/>
          <FormFieldError
            field={{ invalid: true }}
            validations={{ response: 'Madness? This is Sparta!' }}/>
        </FormFieldset>

        <FormFooter>
          <Button submit type="primary" onClick={this.handleSubmit}>
            Submit
          </Button>
        </FormFooter>
      </Form>
    );
  }
}
```
