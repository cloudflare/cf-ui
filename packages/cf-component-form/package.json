{
  "name": "cf-component-form",
  "description": "Cloudflare Form Component",
  "version": "4.2.1",
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js",
  "author": "James Kyle <jkyle@cloudflare.com>",
  "license": "BSD-3-Clause",
  "publishConfig": {
    "registry": "http://registry.npmjs.org/"
  },
  "peerDependencies": {
    "react": "^0.14.9 || ^15.0.0-0 || ^16.0.0-0"
  },
  "dependencies": {
    "cf-style-container": "^1.0.0",
    "polished": "^1.0.0",
    "prop-types": "^15.5.8"
  }
}
