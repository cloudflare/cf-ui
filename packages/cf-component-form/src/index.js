import FormUnstyled from './Form';
import FormTheme from './FormTheme';
import FormHeader from './FormHeader';
import FormFooterUnstyled from './FormFooter';
import FormFooterTheme from './FormFooterTheme';
import FormFieldsetUnstyled from './FormFieldset';
import FormFieldsetTheme from './FormFieldsetTheme';
import FormLabel from './FormLabel';
import FormFieldErrorUnstyled from './FormFieldError';
import FormFieldErrorTheme from './FormFieldErrorTheme';

import { applyTheme } from 'cf-style-container';

const Form = applyTheme(FormUnstyled, FormTheme);
const FormFieldError = applyTheme(FormFieldErrorUnstyled, FormFieldErrorTheme);
const FormFieldset = applyTheme(FormFieldsetUnstyled, FormFieldsetTheme);
const FormFooter = applyTheme(FormFooterUnstyled, FormFooterTheme);

export {
  Form,
  FormUnstyled,
  FormTheme,
  FormHeader,
  FormFooter,
  FormFooterUnstyled,
  FormFooterTheme,
  FormFieldset,
  FormFieldsetUnstyled,
  FormFieldsetTheme,
  FormLabel,
  FormFieldError,
  FormFieldErrorUnstyled,
  FormFieldErrorTheme
};
