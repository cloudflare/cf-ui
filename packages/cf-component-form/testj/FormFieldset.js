import React from 'react';
import renderer from 'react-test-renderer';
import { FormFieldset } from 'cf-component-form';

test('should render', () => {
  const component = renderer.create(
    <FormFieldset legend="Legend">FormFieldset</FormFieldset>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
