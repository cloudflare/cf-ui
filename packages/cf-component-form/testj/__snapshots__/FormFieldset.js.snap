// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<fieldset
  className="cf-form__fieldset"
>
  <legend
    className="cf-form__fieldset_legend"
  >
    Legend
  </legend>
  <div
    className="cf-form__fieldset_content"
  >
    FormFieldset
  </div>
</fieldset>
`;
